packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -50,12 +68,10 @@ class AndroidStudio implements Comparable<AndroidStudio> {
     if (versionString != null)
       version = Version.parse(versionString);
 
-    final String plistValue = iosWorkflow.getPlistValueFromFile(
-      plistFile,
-      null,
-    );
-    final String pathsSelectorValue = _pathsSelectorMatcher.stringMatch(plistValue).split('=').last.trim().replaceAll('"', '');
-    return AndroidStudio(studioPath, version: version, pathsSelectorPath: pathsSelectorValue);
+    final String pathsSelectorValue = extractStudioPlistValueWithMatcher(plistValue, _pathsSelectorMatcher);
+    final String presetPluginsPath = pathsSelectorValue == null ? null :
```

Comment:
small nit: line breaking at places with somewhat symmetrical hierarchy makes it easier to read. Such as

```dart
final String presetPluginsPath = pathsSelectorValue == null
    ? null
    : fs.path.join(homeDirPath, 'Library', 'Application Support', '$pathsSelectorValue');
```